x-pack/plugins/fleet/dev_docs/saved_object_types.md

```diff
@@ -0,0 +1,150 @@
+# Saved Object types
+
+The Fleet plugin leverages several Saved Object types to track metadata on install packages, agent policies, and more.
+This document is intended to outline what each type is for, the primary places it accessed from in the codebase, and
+any caveats regarding the history of that saved object type.
+
+## Types
+
+At this point in time, all times are currently:
+- `hidden: false`
+- `namespaceType: agnostic`
+- `management.importableAndExportable: false`
+
+### `ingest_manager_settings`
+
+- Constant in code: `GLOBAL_SETTINGS_SAVED_OBJECT_TYPE`
+- Introduced in ?
+- Migrations: 7.10.0, 7.13.0
+- [Code Link](../server/saved_objects/index.ts#57)
+
+Tracks the Fleet server host addresses and whether or not the cluster has been shown the "add data" and
+"fleet migration" notices in the UI.
+
+Can be accessed via the APIs exposed in the [server's settings service](../server/services/settings.ts).
+
+### `fleet-agents`
```

Comment:
This one is not used anymore, we did not delete the saved object type as we have a migration that transform that into a `.fleet-agents` document.

Reply:
Got it, makes sense!